tests/Spec.hs

```diff
@@ -621,6 +657,57 @@ main = hspec $ do
       actions   `shouldBe` [ATryPromote (Branch "results/rachael") (Sha "38d")]
 
+    it "pushes and tags with a new version after a successful build" $ do
+      let
+        pullRequest = PullRequest
+          { Project.branch              = Branch "results/rachael"
+          , Project.sha                 = Sha "f35"
+          , Project.title               = "Add my test results"
+          , Project.author              = "rachael"
+          , Project.approval            = Just (Approval "deckard" Project.MergeAndTag)
+          , Project.integrationStatus   = Project.Integrated (Sha "38d") Project.BuildSucceeded
+          , Project.integrationAttempts = []
+          , Project.needsFeedback       = False
+          }
+        state = ProjectState
+          { Project.pullRequests         = IntMap.singleton 1 pullRequest
+          , Project.integrationCandidate = Just $ PullRequestId 1
+          }
+        results = defaultResults { resultIntegrate = [Right (Sha "38e")] }
+        (state', actions) = runActionCustom results $ Logic.proceedUntilFixedPoint state
+        candidate = Project.getIntegrationCandidate state'
+      -- After a successful push, the candidate should be gone.
+      candidate `shouldBe` Nothing
+      actions   `shouldBe` [ ATryPromoteWithTag (Branch "results/rachael") (Sha "38d") (TagName "v2")
+                           , ALeaveComment (PullRequestId 1) "@deckard I tagged your PR with `v2`. Don't forget to deploy it!"
+                           ]
+
+    it "pushes after successful build even if tagging failed" $ do
+      let
+        pullRequest = PullRequest
+          { Project.branch              = Branch "results/rachael"
+          , Project.sha                 = Sha "f35"
+          , Project.title               = "Add my test results"
+          , Project.author              = "rachael"
+          , Project.approval            = Just (Approval "deckard" Project.MergeAndTag)
+          , Project.integrationStatus   = Project.Integrated (Sha "38d") Project.BuildSucceeded
+          , Project.integrationAttempts = []
+          , Project.needsFeedback       = False
+          }
+        state = ProjectState
+          { Project.pullRequests         = IntMap.singleton 1 pullRequest
+          , Project.integrationCandidate = Just $ PullRequestId 1
+          }
+        results = defaultResults { resultIntegrate = [Right (Sha "38e")]
+                                 , resultGetLatestVersion = [Left (TagName "abcdef")] }
+        (state', actions) = runActionCustom results $ Logic.proceedUntilFixedPoint state
+        candidate = Project.getIntegrationCandidate state'
+      -- After a successful push, the candidate should be gone.
+      candidate `shouldBe` Nothing
+      actions   `shouldBe` [ ALeaveComment (PullRequestId 1) "@deckard Sorry, I could not tag your PR. The previous tag `abcdef` seems invalid"
+                           , ATryPromote (Branch "results/rachael") (Sha "38d")]
```

Comment:
Ah, so if tagging fails it will still successfully merge to master? In that case, it's not too bad if we only look at the latest. Makes it somewhat simpler.

Reply:
Looks like someone didn't read the PR description :unamused: Joking aside - this was Hackathon's rush so I just devised something simple to start with. But then I convinced myself that this could be a good idea in the long run. So we never fail to promote because of tagging and in any case you can do that manually if there are problems. OTOH extending it to parse `n` previous tags would not be hard.

Reply:
So - your call :smile:

Reply:
Let's keep it this way then, these situations with broken tags are rare enough.